base_layer/core/src/blocks/block_header.rs

```diff
@@ -134,12 +134,11 @@ impl BlockHeader {
     }
 
     /// Create a new block header using relevant data from the previous block. The height is incremented by one, the
-    /// previous block hash is set, and the timestamp is set to the current time and the proof of work is partially
-    /// initialized, although the `accumulated_difficulty_<algo>` stats are updated using the previous block's proof
-    /// of work information.
-    pub fn from_previous(prev: &BlockHeader) -> Result<BlockHeader, BlockHeaderValidationError> {
+    /// previous block hash is set, the timestamp is set to the current time, and the kernel/output mmr sizes are set to
+    /// the previous block. All other fields, including proof of work are set to defaults.
+    pub fn from_previous(prev: &BlockHeader) -> BlockHeader {
```

Comment:
Is there a reason we don't calculate the accumulated pow data here?
We have the previous block's accumulated pow data and its achieved pow data.

Reply:
We don't have the achieved here AFAICS, that happens after mining and this function is used to supply the block to mine. Though could be a good idea for Chain header maybe.